source/reference/minio-cli/minio-mc/mc-replicate.rst

```diff
@@ -15,46 +15,659 @@ Description
+                        "s3:GetBucketLocation",
+                        "s3:GetBucketVersioning",
+                        "s3:GetBucketObjectLockConfiguration"
+                     ],
+                     "Resource": [
+                        "arn:aws:s3:::DESTINATIONBUCKETNAME"
+                     ]
+               },
+               {
+                     "Effect": "Allow",
+                     "Action": [
+                        "s3:GetReplicationConfiguration",
+                        "s3:ReplicateTags",
+                        "s3:AbortMultipartUpload",
+                        "s3:GetObject",
+                        "s3:GetObjectVersion",
+                        "s3:GetObjectVersionTagging",
+                        "s3:PutObject",
+                        "s3:DeleteObject",
+                        "s3:ReplicateObject",
+                        "s3:ReplicateDelete"
+                     ],
+                     "Resource": [
+                        "arn:aws:s3:::DESTINATIONBUCKETNAME/*"
+                     ]
+               }
+            ]
+         }
+
+      Replace ``DESTINATIONBUCKETNAME`` with the name of the target bucket to
+      which MinIO replicates objects.
+
+      Use the :mc-cmd:`mc admin policy set` command to associate the policy 
+      to a user on the target MinIO cluster.
+
+MinIO strongly recommends creating users specifically for supporting 
+bucket replication operations. See 
+:mc:`mc admin user` and :mc:`mc admin policy` for more complete
+documentation on adding users and policies to a MinIO cluster.
+
+Replication of Existing Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO performs replication as part of writing an object (PUT operations). 
+MinIO does *not* apply replication rules to existing objects written
+*before* enabling replication.
+
+For buckets with existing objects, consider using :mc:`mc mirror` or 
+:mc:`mc cp` to seed the destination bucket. Consider scheduling a maintenance
+window during which applications stop writes to the bucket. Once 
+the :mc:`~mc mirror` or :mc:`~mc cp` fully sync the source and destination,
+enable bucket replication and resume normal operations on the bucket.
+
+Replication of Delete Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports replicating delete operations onto the target bucket. 
+Specifically, MinIO can replicate both 
+:s3-docs:`Delete Markers <versioning-workflows.html>` *and* the deletion
+of specific versioned objects:
+
+- For delete operations on an object, MinIO replication also creates the delete
+  marker on the target bucket.
+
+- For delete operations on versions of an object,
+  MinIO replication also deletes those versions on the target bucket.
+
+MinIO requires explicitly enabling replication of delete operations using the
+:mc-cmd-option:`mc replicate add replicate` flag. This procedure includes the
+required flags for enabling replication of delete operations and delete markers.
+
+.. note::
```

Comment:
@ebozduman so this is not technically related to the mc rm fixups, but I figured I'd ask you what you think of this note. If this captures the core issue, I can apply it wherever we have an op that runs REMOVE under the hood.

Reply:
LGTM

Reply:
Great - I'll actually likely replace it with the note we spoke about on slack DM, but its good to be in the right direction :)